openhands/runtime/impl/remote/remote_runtime.py

```diff
@@ -367,10 +365,10 @@ def _send_action_server_request(self, method, url, **kwargs):
         except requests.Timeout:
             self.log('error', 'No response received within the timeout period.')
             raise
-        except RequestHTTPError as e:
+        except requests.HTTPError as e:
             if e.response.status_code in (404, 502):
-                raise AgentRuntimeDisconnectedError(
-                    f'{e.response.status_code} error while connecting to {self.runtime_url}'
+                raise AgentRuntimeNotFoundError(
```

Comment:
I think `Disconnected` is the right error here, at least for 502--the runtime exists it's just not ready to handle requests

Reply:
fixed!